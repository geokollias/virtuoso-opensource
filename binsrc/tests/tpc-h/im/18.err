Starting Impala Shell without Kerberos authentication
Connected to ip-10-0-1-251.ec2.internal:21000
Server version: impalad version 2.1.3-cdh5 RELEASE (build 20816e26a150d20d2c92c470aa40b342521b243e)
Query: use TPCH
Query: select c_name,
 c_custkey,
 o_orderkey,
 o_orderdate,
 o_totalprice,
 sum(l_quantity)
from
 customer,
 orders,
 lineitem
where
 o_orderkey in (
  select
   l_orderkey
  from
   lineitem
  group by
   l_orderkey having
    sum(l_quantity) > 312
 )
 and c_custkey = o_custkey
 and o_orderkey = l_orderkey
group by
 c_name,
 c_custkey,
 o_orderkey,
 o_orderdate,
 o_totalprice
order by
 o_totalprice desc,
 o_orderdate -- option (loop exists)
limit 100
Fetched 100 row(s) in 412.43s
