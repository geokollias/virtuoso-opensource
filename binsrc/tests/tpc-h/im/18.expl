[?1034h+------------------------------------------------------------------------------------+
| Explain String                                                                     |
+------------------------------------------------------------------------------------+
| Estimated Per-Host Requirements: Memory=6.73GB VCores=4                            |
| WARNING: The following tables are missing relevant table and/or column statistics. |
| tpch.customer                                                                      |
|                                                                                    |
| 17:MERGING-EXCHANGE [UNPARTITIONED]                                                |
| |  order by: o_totalprice DESC, o_orderdate ASC                                    |
| |  limit: 100                                                                      |
| |  hosts=1 per-host-mem=unavailable                                                |
| |  tuple-ids=7 row-size=60B cardinality=100                                        |
| |                                                                                  |
| 09:TOP-N [LIMIT=100]                                                               |
| |  order by: o_totalprice DESC, o_orderdate ASC                                    |
| |  hosts=1 per-host-mem=5.86KB                                                     |
| |  tuple-ids=7 row-size=60B cardinality=100                                        |
| |                                                                                  |
| 16:AGGREGATE [FINALIZE]                                                            |
| |  output: sum:merge(l_quantity)                                                   |
| |  group by: c_name, c_custkey, o_orderkey, o_orderdate, o_totalprice              |
| |  hosts=1 per-host-mem=128.00MB                                                   |
| |  tuple-ids=6 row-size=60B cardinality=79427255                                   |
| |                                                                                  |
| 15:EXCHANGE [HASH(c_name,c_custkey,o_orderkey,o_orderdate,o_totalprice)]           |
| |  hosts=1 per-host-mem=0B                                                         |
| |  tuple-ids=6 row-size=60B cardinality=79427255                                   |
| |                                                                                  |
| 08:AGGREGATE                                                                       |
| |  output: sum(l_quantity)                                                         |
| |  group by: c_name, c_custkey, o_orderkey, o_orderdate, o_totalprice              |
| |  hosts=1 per-host-mem=128.00MB                                                   |
| |  tuple-ids=6 row-size=60B cardinality=79427255                                   |
| |                                                                                  |
| 07:HASH JOIN [LEFT SEMI JOIN, BROADCAST]                                           |
| |  hash predicates: o_orderkey = l_orderkey                                        |
| |  hosts=1 per-host-mem=190.20MB                                                   |
| |  tuple-ids=2,1,0 row-size=72B cardinality=79427255                               |
| |                                                                                  |
| |--14:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=4 row-size=16B cardinality=11331838                                |
| |  |                                                                               |
| |  13:AGGREGATE [FINALIZE]                                                         |
| |  |  output: sum:merge(l_quantity)                                                |
| |  |  group by: l_orderkey                                                         |
| |  |  having: sum(l_quantity) > 312                                                |
| |  |  hosts=1 per-host-mem=190.20MB                                                |
| |  |  tuple-ids=4 row-size=16B cardinality=11331838                                |
| |  |                                                                               |
| |  12:EXCHANGE [HASH(l_orderkey)]                                                  |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=4 row-size=16B cardinality=113318384                               |
| |  |                                                                               |
| |  04:AGGREGATE                                                                    |
| |  |  output: sum(l_quantity)                                                      |
| |  |  group by: l_orderkey                                                         |
| |  |  hosts=1 per-host-mem=1.86GB                                                  |
| |  |  tuple-ids=4 row-size=16B cardinality=113318384                               |
| |  |                                                                               |
| |  03:SCAN HDFS [tpch.lineitem, RANDOM]                                            |
| |     partitions=1/1 files=82 size=20.20GB                                         |
| |     table stats: 600037902 rows total                                            |
| |     column stats: all                                                            |
| |     hosts=1 per-host-mem=176.00MB                                                |
| |     tuple-ids=3 row-size=16B cardinality=600037902                               |
| |                                                                                  |
| 06:HASH JOIN [INNER JOIN, BROADCAST]                                               |
| |  hash predicates: o_custkey = c_custkey                                          |
| |  hosts=1 per-host-mem=314.71MB                                                   |
| |  tuple-ids=2,1,0 row-size=72B cardinality=794272581                              |
| |                                                                                  |
| |--11:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=0 row-size=20B cardinality=15000000                                |
| |  |                                                                               |
| |  00:SCAN HDFS [tpch.customer, RANDOM]                                            |
| |     partitions=1/1 files=5 size=1.18GB                                           |
| |     table stats: 15000000 rows total                                             |
| |     columns missing stats: c_name                                                |
| |     hosts=1 per-host-mem=176.00MB                                                |
| |     tuple-ids=0 row-size=20B cardinality=15000000                                |
| |                                                                                  |
| 05:HASH JOIN [INNER JOIN, BROADCAST]                                               |
| |  hash predicates: l_orderkey = o_orderkey                                        |
| |  hosts=1 per-host-mem=5.53GB                                                     |
| |  tuple-ids=2,1 row-size=52B cardinality=794272581                                |
| |                                                                                  |
| |--10:EXCHANGE [BROADCAST]                                                         |
| |  |  hosts=1 per-host-mem=0B                                                      |
| |  |  tuple-ids=1 row-size=36B cardinality=150000000                               |
| |  |                                                                               |
| |  01:SCAN HDFS [tpch.orders, RANDOM]                                              |
| |     partitions=1/1 files=25 size=6.02GB                                          |
| |     table stats: 150000000 rows total                                            |
| |     column stats: all                                                            |
| |     hosts=1 per-host-mem=352.00MB                                                |
| |     tuple-ids=1 row-size=36B cardinality=150000000                               |
| |                                                                                  |
| 02:SCAN HDFS [tpch.lineitem, RANDOM]                                               |
|    partitions=1/1 files=82 size=20.20GB                                            |
|    table stats: 600037902 rows total                                               |
|    column stats: all                                                               |
|    hosts=1 per-host-mem=176.00MB                                                   |
|    tuple-ids=2 row-size=16B cardinality=600037902                                  |
+------------------------------------------------------------------------------------+
