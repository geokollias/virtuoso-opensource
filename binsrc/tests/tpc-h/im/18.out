[?1034h+--------------------+-----------+------------+---------------------+-------------------+-----------------+
| c_name             | c_custkey | o_orderkey | o_orderdate         | o_totalprice      | sum(l_quantity) |
+--------------------+-----------+------------+---------------------+-------------------+-----------------+
| Customer#011472112 | 11472112  | 458304292  | 1998-02-05 00:00:00 | 591036.15         | 322             |
| Customer#012090925 | 12090925  | 501322081  | 1995-02-04 00:00:00 | 586945.4399999999 | 319             |
| Customer#013458721 | 13458721  | 333307747  | 1997-12-19 00:00:00 | 572334.88         | 319             |
| Customer#008643083 | 8643083   | 84927619   | 1997-06-29 00:00:00 | 571417.48         | 316             |
| Customer#010543705 | 10543705  | 163142919  | 1996-06-10 00:00:00 | 569798.1          | 313             |
| Customer#005914657 | 5914657   | 55799200   | 1996-02-11 00:00:00 | 568754.48         | 327             |
| Customer#011461310 | 11461310  | 20662370   | 1992-07-22 00:00:00 | 566353.2          | 315             |
| Customer#007682894 | 7682894   | 532736640  | 1993-07-22 00:00:00 | 563532.16         | 316             |
| Customer#012070685 | 12070685  | 34201984   | 1997-06-18 00:00:00 | 563492.49         | 322             |
| Customer#012878113 | 12878113  | 42290181   | 1997-11-26 00:00:00 | 563479.5699999999 | 318             |
| Customer#000506836 | 506836    | 116567399  | 1992-03-21 00:00:00 | 563342.71         | 319             |
| Customer#012848143 | 12848143  | 501771233  | 1994-12-06 00:00:00 | 562642.03         | 317             |
| Customer#003493022 | 3493022   | 182898470  | 1996-01-09 00:00:00 | 561296.0600000001 | 318             |
| Customer#013083751 | 13083751  | 28077922   | 1996-10-27 00:00:00 | 560893.88         | 319             |
| Customer#006580315 | 6580315   | 343593507  | 1994-05-20 00:00:00 | 560058.6          | 321             |
| Customer#009427249 | 9427249   | 48881602   | 1993-09-19 00:00:00 | 559502.8199999999 | 327             |
| Customer#006594658 | 6594658   | 555833473  | 1997-08-01 00:00:00 | 557961.99         | 324             |
| Customer#005925541 | 5925541   | 446215044  | 1992-05-13 00:00:00 | 556993.03         | 318             |
| Customer#007010650 | 7010650   | 587222276  | 1993-12-17 00:00:00 | 555946.71         | 329             |
| Customer#008211575 | 8211575   | 218305350  | 1992-12-19 00:00:00 | 555691.15         | 316             |
| Customer#005523215 | 5523215   | 499594150  | 1997-12-30 00:00:00 | 552295.59         | 322             |
| Customer#006048598 | 6048598   | 45897379   | 1994-05-30 00:00:00 | 550195.48         | 318             |
| Customer#013889867 | 13889867  | 322196069  | 1992-12-24 00:00:00 | 549996.0600000001 | 313             |
| Customer#003262783 | 3262783   | 327814241  | 1997-04-16 00:00:00 | 548355.89         | 319             |
| Customer#002099413 | 2099413   | 507968289  | 1993-01-23 00:00:00 | 547809.74         | 316             |
| Customer#005800261 | 5800261   | 493616454  | 1997-01-11 00:00:00 | 547151.89         | 314             |
| Customer#012179552 | 12179552  | 157723879  | 1992-01-06 00:00:00 | 545008.9300000001 | 320             |
| Customer#000328568 | 328568    | 584293472  | 1997-02-04 00:00:00 | 544997.41         | 317             |
| Customer#005098117 | 5098117   | 82043264   | 1996-04-15 00:00:00 | 544838.33         | 330             |
| Customer#006394964 | 6394964   | 410876964  | 1994-11-15 00:00:00 | 543654.37         | 324             |
| Customer#005021956 | 5021956   | 163897538  | 1992-05-10 00:00:00 | 543411.84         | 315             |
| Customer#014745380 | 14745380  | 323533249  | 1993-03-31 00:00:00 | 543411.8199999999 | 322             |
| Customer#001070318 | 1070318   | 91475014   | 1996-11-15 00:00:00 | 542874.97         | 314             |
| Customer#000793051 | 793051    | 103963585  | 1996-10-27 00:00:00 | 540445.27         | 313             |
| Customer#010763563 | 10763563  | 575451169  | 1994-01-18 00:00:00 | 539403.5600000001 | 329             |
| Customer#008402254 | 8402254   | 539392967  | 1993-02-23 00:00:00 | 538886.3100000001 | 333             |
| Customer#008654869 | 8654869   | 100176839  | 1993-04-12 00:00:00 | 538846.61         | 317             |
| Customer#014286853 | 14286853  | 423877508  | 1995-03-02 00:00:00 | 536949.1800000001 | 319             |
| Customer#008779822 | 8779822   | 211965092  | 1996-07-18 00:00:00 | 536768.4300000001 | 314             |
| Customer#008208034 | 8208034   | 446989058  | 1993-07-12 00:00:00 | 536601.61         | 315             |
| Customer#009666952 | 9666952   | 475794275  | 1994-09-23 00:00:00 | 536427.97         | 322             |
| Customer#014831182 | 14831182  | 92086567   | 1996-06-12 00:00:00 | 536294.92         | 322             |
| Customer#013677323 | 13677323  | 387783079  | 1992-04-22 00:00:00 | 535504.63         | 316             |
| Customer#004663856 | 4663856   | 159914310  | 1996-09-12 00:00:00 | 535322.28         | 315             |
| Customer#000299162 | 299162    | 303098309  | 1996-01-28 00:00:00 | 534421.58         | 315             |
| Customer#005747806 | 5747806   | 540886884  | 1994-04-14 00:00:00 | 534288.51         | 314             |
| Customer#013094119 | 13094119  | 111128577  | 1995-05-09 00:00:00 | 534239.89         | 316             |
| Customer#012726752 | 12726752  | 92511430   | 1995-03-06 00:00:00 | 533441.05         | 318             |
| Customer#011219830 | 11219830  | 156065860  | 1996-07-03 00:00:00 | 533119.17         | 325             |
| Customer#014112568 | 14112568  | 280056096  | 1997-01-30 00:00:00 | 532881.87         | 313             |
| Customer#013477537 | 13477537  | 175859779  | 1997-04-14 00:00:00 | 532574.3          | 325             |
| Customer#003082444 | 3082444   | 319083271  | 1993-03-24 00:00:00 | 532223.76         | 315             |
| Customer#010716088 | 10716088  | 275341121  | 1994-07-15 00:00:00 | 532037.67         | 326             |
| Customer#009444602 | 9444602   | 177770599  | 1992-01-28 00:00:00 | 531961.4          | 337             |
| Customer#013616422 | 13616422  | 329431840  | 1995-09-05 00:00:00 | 531857.34         | 316             |
| Customer#000155713 | 155713    | 531652866  | 1996-06-18 00:00:00 | 531350.79         | 326             |
| Customer#004808569 | 4808569   | 12809282   | 1994-05-21 00:00:00 | 531002.73         | 314             |
| Customer#013300369 | 13300369  | 559711910  | 1993-03-25 00:00:00 | 530710.84         | 324             |
| Customer#003172843 | 3172843   | 44178691   | 1998-01-24 00:00:00 | 530083.88         | 316             |
| Customer#014675531 | 14675531  | 358712709  | 1996-01-15 00:00:00 | 529299.34         | 314             |
| Customer#007404905 | 7404905   | 87006309   | 1992-11-21 00:00:00 | 529089.99         | 320             |
| Customer#007773805 | 7773805   | 261498084  | 1993-12-24 00:00:00 | 528495.53         | 319             |
| Customer#014160562 | 14160562  | 525435843  | 1994-04-25 00:00:00 | 527900.25         | 313             |
| Customer#009668443 | 9668443   | 327357765  | 1995-02-22 00:00:00 | 527800.9          | 325             |
| Customer#006125536 | 6125536   | 532672775  | 1993-12-16 00:00:00 | 527552.58         | 326             |
| Customer#004041908 | 4041908   | 99667621   | 1992-10-25 00:00:00 | 526326.09         | 317             |
| Customer#006541660 | 6541660   | 130130438  | 1992-03-31 00:00:00 | 525783.23         | 324             |
| Customer#012270997 | 12270997  | 35542498   | 1993-10-12 00:00:00 | 525713.29         | 325             |
| Customer#007789483 | 7789483   | 40876003   | 1992-05-10 00:00:00 | 525433.7          | 314             |
| Customer#002620171 | 2620171   | 546228576  | 1994-11-28 00:00:00 | 524372.13         | 325             |
| Customer#011942315 | 11942315  | 137209030  | 1993-07-14 00:00:00 | 524187.78         | 323             |
| Customer#013455998 | 13455998  | 483724192  | 1997-03-04 00:00:00 | 523481.16         | 314             |
| Customer#007051187 | 7051187   | 138606375  | 1997-07-24 00:00:00 | 522926.52         | 318             |
| Customer#002021647 | 2021647   | 309781410  | 1995-04-12 00:00:00 | 521591.5          | 313             |
| Customer#005861503 | 5861503   | 395809221  | 1996-03-11 00:00:00 | 520971.04         | 314             |
| Customer#002667464 | 2667464   | 41274690   | 1992-08-10 00:00:00 | 520813.13         | 317             |
| Customer#007297621 | 7297621   | 442178406  | 1998-01-30 00:00:00 | 520565.56         | 318             |
| Customer#014874056 | 14874056  | 468303910  | 1998-01-29 00:00:00 | 519999.41         | 314             |
| Customer#012912623 | 12912623  | 187711367  | 1995-08-17 00:00:00 | 519950.45         | 325             |
| Customer#008050909 | 8050909   | 260432385  | 1994-09-13 00:00:00 | 519518.68         | 324             |
| Customer#007354688 | 7354688   | 584032992  | 1992-02-01 00:00:00 | 518933.52         | 316             |
| Customer#010930157 | 10930157  | 232919687  | 1997-05-21 00:00:00 | 518754.11         | 334             |
| Customer#006014731 | 6014731   | 161387302  | 1997-06-19 00:00:00 | 518719.77         | 314             |
| Customer#000280583 | 280583    | 224453441  | 1998-03-10 00:00:00 | 518616.51         | 318             |
| Customer#000040175 | 40175     | 507136258  | 1992-02-01 00:00:00 | 518497.04         | 317             |
| Customer#000358255 | 358255    | 396154496  | 1992-01-17 00:00:00 | 518095.13         | 317             |
| Customer#005907479 | 5907479   | 495435649  | 1992-05-07 00:00:00 | 517990.12         | 313             |
| Customer#008149051 | 8149051   | 127825121  | 1995-04-16 00:00:00 | 517895.56         | 320             |
| Customer#004261654 | 4261654   | 455281089  | 1997-09-08 00:00:00 | 517667.92         | 313             |
| Customer#004493219 | 4493219   | 146084455  | 1992-08-21 00:00:00 | 517549.01         | 313             |
| Customer#009049549 | 9049549   | 323219458  | 1994-04-02 00:00:00 | 517408.68         | 315             |
| Customer#013163174 | 13163174  | 546761091  | 1998-02-06 00:00:00 | 517316.41         | 316             |
| Customer#014141798 | 14141798  | 531081185  | 1996-06-08 00:00:00 | 516399.57         | 321             |
| Customer#011173516 | 11173516  | 143875140  | 1998-06-09 00:00:00 | 516258.94         | 315             |
| Customer#004621336 | 4621336   | 294701219  | 1993-05-10 00:00:00 | 516081.84         | 315             |
| Customer#009433798 | 9433798   | 469098182  | 1997-10-17 00:00:00 | 516024.87         | 317             |
| Customer#011719279 | 11719279  | 162040167  | 1998-05-04 00:00:00 | 515755.61         | 313             |
| Customer#009814556 | 9814556   | 20971013   | 1997-08-12 00:00:00 | 515684.69         | 322             |
| Customer#008605375 | 8605375   | 595511555  | 1993-08-12 00:00:00 | 515645.46         | 314             |
| Customer#014111593 | 14111593  | 233925702  | 1993-08-23 00:00:00 | 515501.24         | 318             |
+--------------------+-----------+------------+---------------------+-------------------+-----------------+
